nova/virt/vmwareapi/driver.py

```diff
@@ -487,6 +488,30 @@ def get_inventory(self, nodename):
                 'max_unit': stats['mem']['max_mem_mb_per_host'],
                 'step_size': 1,
             }})
+
+            # Calculate total reservable memory, depending on either the
+            # configured tolerance for failed hypervisors or, if that setting
+            # is disabled, a ratio of maximum reserved memory to total
+            # available memory.
+            memory_available_total = stats['mem']['total'] - reserved_memory_mb
+            hv_count = stats.get('hypervisor_count', False)
+            max_fail_hvs = \
+                max(0, CONF.vmware.memory_reservation_cluster_hosts_max_fail)
+            if max_fail_hvs and hv_count:
+                total_memory_reservable = \
+                    memory_available_total * (1 - max_fail_hvs / hv_count)
+            else:
+                total_memory_reservable = memory_available_total \
+                    * CONF.vmware.memory_reservation_max_ratio_fallback
```

Comment:
Why do it here instead of in `vm_util.get_stats_from_cluster()`? Any specific reason?

Reply:
Nope. Just that this is where the other resources are filled in, so it seemed as good a place as any. In regards to you other comment below, I will indeed move it into get_stats because then I don't have to pass around hypervisor count, like you pointed out. Probably the diff would be smaller too.